detox/src/ios/system.js

```diff
@@ -0,0 +1,125 @@
+const assert = require('assert');
+
+const { DetoxRuntimeError } = require('../errors');
+const { systemActionDescription, expectDescription } = require('../utils/invocationTraceDescriptions');
+const log = require('../utils/logger').child({ cat: 'ws-client, ws' });
+const traceInvocationCall = require('../utils/traceInvocationCall').bind(null, log);
+
+
+class SystemExpect {
+  constructor(xcuitestRunner, element) {
+    this._xcuitestRunner = xcuitestRunner;
+    this.element = element;
+    this.modifiers = [];
+  }
+
+  toExist() {
+    const traceDescription = expectDescription.toExist();
+    return this.expect('toExist', traceDescription);
+  }
+
+  get not() {
+    this.modifiers.push('not');
+    return this;
+  }
+
+  createInvocation(systemExpectation) {
+    return {
+      type: 'systemExpectation',
+      systemPredicate: this.element.matcher.predicate,
+      ...(this.element.index !== undefined && { systemAtIndex: this.element.index }),
+      ...(this.modifiers.length !== 0 && { systemModifiers: this.modifiers }),
+      systemExpectation
+    };
+  }
+
+  expect(expectation, traceDescription) {
+    assert(traceDescription, `must provide trace description for expectation: \n ${JSON.stringify(expectation)}`);
```

Comment:
I feel uneasy about eager object stringifying each time we try to send something. To my embarrassment, I would even be fine with stupid `if (!traceDescription) { assert(...) }`, just not to spend CPU time for things that won't be needed in 99.99% cases.

I see two good options:
1) check if stack traces are usable even without this stringification – this really can be so
2) if not, use homebrewed assertion utils from `./src/utils` (if needed, add some)

Sorry for nitpicking. 🙏

Reply:
I wonder if some workaround like

```js
const lazyMessage = {
    toString: function() {
        return `... ${JSON.stringify(...)}`;
    }
};
```

will do the trick when passing it to the assert function. however that's a bit dirty.

Reply:
you're right about 1, the JSON is redundant here.. there's enough information without it